openmc/tallies.py

```diff
@@ -452,9 +449,13 @@ def add_score(self, score):
                   'not a string'.format(score, self.id)
             raise ValueError(msg)
 
-        # If the score is already in the Tally, don't add it again
+        # If the score is already in the Tally, raise an error
         if score in self.scores:
-            return
+            msg = 'Unable to add a duplicate score {0} to Tally ID="{1}" ' \
```

Comment:
This is fine by me, but I think we should get @paulromano and/or
@smharper's opinion on this before merging. This kind of "asymmetric"
functionality between XML and Python may be undesirable, unless we conclude
that thorough error checking of inputs in Fortran will bloat
"input_xml.F90."
On Dec 25, 2015 12:38 PM, "sam" <notifications@github.com> wrote:

> In openmc/tallies.py
> <https://github.com/mit-crpg/openmc/pull/534#discussion_r48443241>:
>
> >          if score in self.scores:
> > -            return
> > +            msg = 'Unable to add a duplicate score {0} to Tally ID="{1}" ' \
>
> Correct, only the Python API precludes duplicate scores. Before, the
> Python API was able to read in tally data with duplicate scores, but I
> don't think it would be able to perform tally arithmetic correctly or print
> out the data correctly. Furthermore, the Python API wasn't able to create a
> tallies file with duplicate scores, so the duplicate scores would only be
> encountered when the tallies.xml file was generated without the Python API
> and then the Python API was used to read in the tallies data.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mit-crpg/openmc/pull/534/files#r48443241>.
>